modules/core/test/test_mat.cpp

```diff
@@ -2669,4 +2669,219 @@ TEST(Mat, regression_26322)
     EXPECT_EQ(0, (int)mat3.total());
 }
 
+TEST(Mat, reshape_1d)
+{
+    std::vector<int> v = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11};
+    int newrows = 2;
+    Mat m = Mat(v).reshape(0, newrows);
+
+    EXPECT_EQ(m.dims, newrows);
```

Comment:
It should be just 2, but not `newrows`. They are not correlated.

Reply:
fixed